src/lib/libasync.js

```diff
@@ -476,11 +476,11 @@ addToLibrary({
 
   emscripten_sleep__deps: ['$safeSetTimeout'],
   emscripten_sleep__async: true,
-  emscripten_sleep: (ms) => Asyncify.handleSleep((wakeUp) => safeSetTimeout(wakeUp, ms)),
+  emscripten_sleep: async (ms) => Asyncify.handleSleep((wakeUp) => safeSetTimeout(wakeUp, ms)),
```

Comment:
What does flagging this `async` gain here? (it costs code size)

I understand the rationale for `async` is that one can `await` for the return value, but for functions that return void, isn't the `async` keyword dead code?

Though Asyncify and JSPI build modes itself do not need the functions to be flagged async do they? (in [WebGPU JSPI support I went with this](https://github.com/juj/wasm_webgpu/blob/3ed529956d1ee3f14f64c07769385e1929908f68/lib/lib_webgpu.js#L779-L806))

Reply:
It really doesn't add anything here, but consistency. We have a few spots where we auto modify the library js (e.g. memory 64). It's much easier in those wrapper functions to assume we're in a `async` function and emit `await` code (also the `await` code is usually shorter).